### PR TITLE
refactor (jkube-kit/generator/karaf/): Migrate all tests from JMockit to Mockito

### PR DESCRIPTION
**Pull request by baruKreddy**
## Description

Fixes https://github.com/eclipse/jkube/issues/1595

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/generator/karaf/src/test/java/org/eclipse/jkube/generator/karaf/KarafGeneratorTest.java [hunk 1]
```diff
@@ -26,141 +26,121 @@
...
-        .customize(originalImageConfigurations, false);
+            .customize(originalImageConfigurations, false);
     // Then
-    assertThat(result).singleElement()
-            .extracting(ImageConfiguration::getBuildConfiguration)
-            .hasFieldOrPropertyWithValue("ports", Arrays.asList("8080", "8778"))
-            .extracting(BuildConfiguration::getEnv)
-            .asInstanceOf(InstanceOfAssertFactories.MAP)
-            .containsEntry("DEPLOYMENTS_DIR", "/other-dir");
+    assertThat(result).hasSize(1);
```
Comment (rohanKanojia) [inline jkube-kit/generator/karaf/src/test/java/org/eclipse/jkube/generator/karaf/KarafGeneratorTest.java:141]: Why is this change required? It was already single chained assertion before.

File: jkube-kit/generator/karaf/src/test/java/org/eclipse/jkube/generator/karaf/KarafGeneratorTest.java [hunk 2]
```diff
@@ -26,141 +26,121 @@
...
+                            new File(temporaryFolder, "assembly"),
+                            new File("karaf"),
+                            "0775",
+                            null),
+                    tuple(
+                            temporaryFolder.toPath().resolve("assembly").resolve("bin").toFile(),
+                            new File("karaf", "bin"),
+                            "0775",
+                            "0777")
+            );
```
Comment (rohanKanojia) [inline jkube-kit/generator/karaf/src/test/java/org/eclipse/jkube/generator/karaf/KarafGeneratorTest.java:124]: Could you please revert indentation changes?

**Comment by rohanKanojia** [comment]
@baruKreddy : It looks like you've pushed wrong commit to this branch which is causing compilation failure

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
